cdap-docs/reference-manual/source/glossary.rst

```diff
@@ -154,7 +154,15 @@ Glossary
       enable exploration, you must set several properties when creating the stream or
       dataset, and the files in a dataset must meet certain requirements.
 
-   .. ETL and Application Templates
+   Producer
+      A producer is an entity that emits events, such as a :term:`stream` or a
+      :term:`flowlet`. Flowlets can at the same time be both :term:`consumers <consumer>`
+      and producers.
+
+   Consumer
+      A consumer is an entity that is a recipient of events, typically a :term:`flowlet`.
+      A consumer flowlet is the recipient of a queue. The difference between the total of
+      the :term:`producers <producer>` and the consumers is the pending events.
```

Comment:
difference between the total of produced events and the consumed events is the pending events?